EricssonBroadcastServices/javascript-sdk
# feat(EMP-20239): RBMOTT SDK

[SDK Readme](https://github.com/EricssonBroadcastServices/javascript-sdk/blob/9703e62b0d020acc359de80fa615dd006b782544/packages/rbm-ott-sdk/README.md)


To test the generator:
```sh
npm install
npm run update-spec -w @ericssonbroadcastservices/rbm-ott-sdk # download the exposure api spec locally
npm run generate -w @ericssonbroadcastservices/rbm-ott-sdk # generate
```

I don't feel super confident about the auth and error handling parts, although I have tested both and they work well (Imanaged to migrate ExposureService in js-player). It now checks for `sessionToken` in the response body and if it can find it it saves it and uses it for all the coming requests. This part probably needs some input.

The API JSON spec specifies that some of the endpoints supports the authorization header, and I tested to restrict it to be added for only those (37dca1c), but the spec is incomplete, so I had to revert that, but I think it's also risky to save it the way I do (globally per cu/bu rather than in the context object). I tried to save it in the context object first, but ran into issues with that.

Other than that I would want input for the file names, type/interface names (should they be prefixed/suffixed?). Should the "context" object be named something else?
